cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -633,6 +633,103 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
         }
     });
 
+    UnitAccessEditor = AbstractVisibilityEditor.extend({
+        templateName: 'unit-access-editor',
+        className: 'edit-unit-access',
+        events: {
+            'change .user-partition-select': function() {
+                this.hideCheckboxDivs();
+                this.showSelectedDiv(this.getSelectedEnrollmentTrackId());
+                this.getSelectedGroupAccess();
+            }
+        },
+
+        afterRender: function() {
+            var groupAccess,
+                keys;
+            AbstractVisibilityEditor.prototype.afterRender.call(this);
+            this.hideCheckboxDivs();
+            if (this.model.attributes.group_access) {
+                groupAccess = this.model.attributes.group_access;
+                keys = Object.keys(groupAccess);
+                if (keys.length === 1) { // should be only one partition key
+                    if (groupAccess.hasOwnProperty(keys[0]) && groupAccess[keys[0]].length > 0) {
+                        // Select the option that has group access, provided there is a specific group within the scheme
+                        this.$('.user-partition-select option[value=' + keys[0] + ']').prop('selected', true);
+                        this.showSelectedDiv(keys[0]);
+                        this.$('#partition-select option:contains("Select a group type")').text(
```

Comment:
For your second comment, we're copying the UI from the access modal on the component page, where it says "Select a group type" if you haven't previously selected any group to restrict access to.
We previously had it always say "All Learners and Staff" before we changed it for the last review

@roderickmorales Any opinion on which one we should decide on?